retriever.py

```diff
@@ -0,0 +1,272 @@
+                            )
+            image_io = io.BytesIO(img.read())
+            img = PIL.Image.open(image_io)
+
+            bm.imshow(img, interpolation="nearest", origin="upper")
+            bm.drawcoastlines()
+            bm.drawcountries()
+
+            fig.savefig(f"{flight}_{layer}.png")
+
+        # prepare vsec plots
+        path = [(wp[0], wp[1], datetime.datetime.now()) for wp in wps]
+        lats, lons = mslib.utils.coordinate.path_points(
+            [_x[0] for _x in path], [_x[1] for _x in path], numpoints=num_interpolation_points + 1, connection="greatcircle")
+        intermediate_indexes = []
+        ipoint = 0
+        for i, (lat, lon) in enumerate(zip(lats, lons)):
+            if abs(lat - wps[ipoint][0]) < 1E-10 and abs(lon - wps[ipoint][1]) < 1E-10:
+                intermediate_indexes.append(i)
+                ipoint += 1
+            if ipoint >= len(wps):
+                break
+
+        for url, layer, style in config["automated_plotting_vsecs"]:
+            fig.clear()
+
+            # setup ticks and labels
+            ax = fig.add_subplot(111, zorder=99)
+            ax.set_yscale("log")
+            p_bot, p_top = [float(x) * 100 for x in vertical.split(",")]
+            bbox = ",".join(str(x) for x in (num_interpolation_points, p_bot / 100, num_labels, p_top / 100))
+            ax.grid(b=True)
+            ax.patch.set_facecolor("None")
+            pres_maj = mslib.msui.mpl_qtwidget.MplSideViewCanvas._pres_maj
+            pres_min = mslib.msui.mpl_qtwidget.MplSideViewCanvas._pres_min
+            major_ticks = pres_maj[(pres_maj <= p_bot) & (pres_maj >= p_top)]
+            minor_ticks = pres_min[(pres_min <= p_bot) & (pres_min >= p_top)]
+            labels = ["{}".format(int(l / 100.))
+                      if (l / 100.) - int(l / 100.) == 0 else "{}".format(float(l / 100.)) for l in major_ticks]
+            if len(labels) > 20:
+                labels = ["" if x.split(".")[-1][0] in "975" else x for x in labels]
+            elif len(labels) > 10:
+                labels = ["" if x.split(".")[-1][0] in "9" else x for x in labels]
+            ax.set_ylabel("pressure (hPa)")
+            ax.set_yticks(minor_ticks, minor=True)
+            ax.set_yticks(major_ticks, minor=False)
+            ax.set_yticklabels([], minor=True, fontsize=10)
+            ax.set_yticklabels(labels, minor=False, fontsize=10)
+            ax.set_ylim(p_bot, p_top)
+            ax.set_xlim(0, num_interpolation_points)
+            ax.set_xticks(range(0, num_interpolation_points, tick_index_step))
+            ax.set_xticklabels(
+                [f"{x[0]:2.1f}, {x[1]:2.1f}"
+                 for x in zip(lats[::tick_index_step], lons[::tick_index_step])],
+                rotation=25, fontsize=10, horizontalalignment="right")
+            ax.set_xlabel("lat/lon")
+
+            # plot path and waypoint labels
+            ax.plot(intermediate_indexes, wp_presss,
```

Comment:
That is now from the VPathInteractor class from mslib/msui/mpl_pathinteractor.py.
In particular from method redraw_path from lines 520ff.